connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerSinkTaskTest.java

```diff
@@ -348,6 +350,30 @@ public void testWakeupInCommitSyncCausesRetry() throws Exception {
         PowerMock.verifyAll();
     }
 
+    @Test
+    public void testTimestampPropagation() throws Exception {
+        expectInitializeTask();
+        expectConsumerPoll(1);
+        expectConvertMessages(1);
+
+        Capture<Collection<SinkRecord>> records = EasyMock.newCapture(CaptureType.ALL);
+
+        sinkTask.put(EasyMock.capture(records));
+
+        PowerMock.replayAll();
+
+        workerTask.initialize(TASK_CONFIG);
+        workerTask.initializeAndStart();
+        workerTask.poll(Long.MAX_VALUE);
+
+        SinkRecord record = records.getValue().iterator().next();
+
+        assertEquals(TIMESTAMP, record.timestamp());
```

Comment:
Not sure how easy this is with current test code setup, but it seems like we might need to verify translation of a couple of different timestamps, i.e. at least adding the case of no timestamp. I think this is just going to be annoying since `expectConsumerPoll` currently uses a bunch of fixed values.